internal/runtime/catalog/test/v1alpha1/conversion_test.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog/test/v1alpha2"
+)
+
+func TestConversion(t *testing.T) {
+	g := NewWithT(t)
+
+	var c = catalog.New()
+	_ = AddToCatalog(c)
+	_ = v1alpha2.AddToCatalog(c)
+
+	t.Run("down-convert FakeRequest v1alpha2 to v1alpha1", func(t *testing.T) {
+		oldRequest := &v1alpha2.FakeRequest{Cluster: clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{
```

Comment:
ups sorry, I meant old in the sense of v1alpha1 new = v1alpha2

Reply:
So maybe old/new is not so great. Totally up to you what to do with that comment, you can just leave it as is